libsolidity/ast/AST.h

```diff
@@ -1183,17 +1204,16 @@ class ElementaryTypeName: public TypeName
 class UserDefinedTypeName: public TypeName
 {
 public:
-	UserDefinedTypeName(int64_t _id, SourceLocation const& _location, std::vector<ASTString> _namePath):
+	UserDefinedTypeName(int64_t _id, SourceLocation const& _location, ASTPointer<IdentifierPath> _namePath):
 		TypeName(_id, _location), m_namePath(std::move(_namePath)) {}
 	void accept(ASTVisitor& _visitor) override;
 	void accept(ASTConstVisitor& _visitor) const override;
 
-	std::vector<ASTString> const& namePath() const { return m_namePath; }
-
-	UserDefinedTypeNameAnnotation& annotation() const override;
+	std::vector<ASTString> const& namePath() const { return m_namePath->path(); }
+	ASTPointer<IdentifierPath> const& pathNode() const { return m_namePath; }
```

Comment:
Wouldn't it be better to return `*m_namePath`? It should not be allowed to be null - we can add an assert in the constructor.